README.md

```diff
@@ -1,78 +1,119 @@
+When a task update changes the interface (e.g., change of parameters, workspaces or results names), a new version of the task should be created. The folder with the new version must contain `MIGRATION.md` with instructions on how to update the current pipeline file in user's `.tekton` folder.
 
-Adding a new parameter with a default value does not require the task version increase.
+Adding a new parameter with a default value does not require a task version increase.
 
-Task version increase must be approved by Project Manager.
+Task version increase must be approved by the Project Manager.
 
-## Testing
+## Local development
+Tasks can have a TA (Trusted Artifact) version.
+The recommended workflow is to only edit the base version and let the other versions get generated automatically.
+```
+./hack/generate-ta-tasks.sh
+```
+Buildah also has a remote version, which can be generated with:
+```
+./hack/generate-buildah-remote.sh
+```
 
-Script `./hack/test-builds.sh` creates pipelines and tasks directly in current namespace and executes there test builds. By setting the environment variable `QUAY_NAMESPACE` the images will be pushed into user's quay repository, in that case creation of secret named `redhat-appstudio-staginguser-pull-secret` is required.
+## Testing
 
-Script `./hack/test-build.sh` provides way to test on custom git repository and pipeline. Usage example: `./hack/test-build.sh https://github.com/jduimovich/spring-petclinic java-builder`.
+### Prerequisites
+- Provisioned cluster with sufficient resources
+- Deployed Konflux on the cluster (see [infra-deployments](https://github.com/redhat-appstudio/infra-deployments)) 
+
+1. Set up the image repository
+PipelineRuns attempt to push to `registry.redhat.io` by default. 
+For testing, you will likely want to use your own Quay repository. 
+Specify the Quay repository using the `QUAY_NAMESPACE` environment variable in the format `OWNER/REPOSITORY_NAME`.
+2. Set up the `redhat-appstudio-staginguser-pull-secret`
+   - Log in to `quay.io` using your credentials:
+     ```
+     podman login quay.io
+     ```
+     This will create an `auth.json` file in `${XDG_RUNTIME_DIR}/containers/auth.json`, which you will use to create a secret in the cluster.
+   - Create the pull secret in you cluster:
+     ```
+     oc create secret docker-registry redhat-appstudio-staginguser-pull-secret --from-file=.dockerconfigjson=${XDG_RUNTIME_DIR}/containers/auth.json
+     ```
+   - Link the secret to your service account:
+     ```
+     oc secrets link appstudio-pipeline redhat-appstudio-staginguser-pull-secret
+     ```
+3. Run the tests
+
+- To test a custom Git repository and pipeline, use `./hack/test-build.sh`. 
+  
+  Usage example:
+  ```
+  ./hack/test-build.sh https://github.com/jduimovich/spring-petclinic java-builder`.
```

Comment:
Would I have to do something special to use my own quay repository? Maybe invoke this with `QUAY_NAMESPACE=my-quay-namespace ./hack/test-build.sh ...`?